cmd/juju/application/deploy.go

```diff
@@ -385,6 +385,8 @@ type DeployCommand struct {
 
 	machineMap string
 	flagSet    *gnuflag.FlagSet
+
+	notLocalModel bool
```

Comment:
I'm not sure notLocalModel is the best name here, perhaps something like notKnownModel or notCachedModel or perhaps even modelKnownAtInitTime (no that's too wordy :-) ) to indicate we're dealing with a model that we don't currently know about.

Reply:
I was originally thinking notKnownLocallyModel but is it really adding more clarity?

Reply:
I'll just rename to unknownModel.